extensions/exquery/restxq/src/main/java/org/exist/extensions/exquery/restxq/impl/RestXqServiceImpl.java

```diff
@@ -297,4 +319,185 @@ private static StringValue parseAsString(final InputStream is, final String enco
         final String s = new String(bos.toByteArray(), encoding);
         return new StringValue(s);
     }
+    
+    /**
+     * Extract parts from multipart/related request and build XML http:multipart Node as String including body parts and headers
+     * The first body element is the root part.
+	 * Body content is represented as BASE64
+	 * 
+     * <http:multipart>
+  	 * 		<http:header name="content-id" value="..." />
+  	 *		<http:body media-type="...">...</http:body>
+	 *		.
+	 *		. 
+	 * 	</http:multipart>
+	 * 
+     */
+    private String extractBodyParts(final HttpRequest request, final InputStream is, final String encoding) throws RestXqServiceException, IOException{
+    	
+    	final String requestContentType = request.getContentType();
+    	final String boundary = getMultiPartBoundary(requestContentType);
+    	final String rootId = getRootPartId(requestContentType);
+		final String content = getRequestContent(is, encoding);
+		
+		// build http:header and http:body for each body part within a http:mulitpart as XML String
+		// @see <a href="http://expath.org/spec/http-client">HTTP Client Module</a>
+		final StringBuilder xmlPartsAsStr = new StringBuilder();
+		
+		// compile part Content-Type pattern before going through the parts
```

Comment:
Please move these out of the methods into class level (performance reasons)

Reply:
That's plausible and i will work on it